src/Test/Plutip/Internal/LocalCluster.hs

```diff
@@ -11,85 +6,66 @@ module Test.Plutip.Internal.LocalCluster (
-  walletListenFromEnv,
-  walletMinSeverityFromEnv,
-  withCluster,
- )
-import Control.Arrow (first)
-import Control.Concurrent (threadDelay)
+
+import Cardano.BM.Data.Severity qualified as Severity
+import Cardano.BM.Data.Tracer (HasPrivacyAnnotation, HasSeverityAnnotation (getSeverityAnnotation))
+import Cardano.CLI (LogOutput (LogToFile, LogToStdStreams), withLoggingNamed)
+import Cardano.Wallet.Shelley.Launch.Cluster (ClusterLog, localClusterConfigFromEnv, testMinSeverityFromEnv, walletMinSeverityFromEnv, withCluster)
 import Control.Concurrent.Async (async)
-import Control.Monad (unless, void, when)
+import Control.Monad (unless, void)
+import Control.Monad.IO.Class (liftIO)
 import Control.Monad.Reader (ReaderT (runReaderT))
+import Control.Retry (constantDelay, limitRetries, recoverAll)
 import Control.Tracer (Tracer, contramap, traceWith)
 import Data.Kind (Type)
-import Data.Maybe (catMaybes, isJust)
-import Data.Proxy (Proxy (..))
+import Data.Maybe (catMaybes, fromMaybe, isJust)
 import Data.Text (Text, pack)
-import Data.Text qualified as T
-import Data.Text.Class (ToText (..))
+import Data.Text.Class (ToText (toText))
 import Plutus.ChainIndex.App qualified as ChainIndex
 import Plutus.ChainIndex.Config (ChainIndexConfig (cicNetworkId, cicPort), cicDbPath, cicSocketPath)
-import Plutus.ChainIndex.Config qualified as CI
-import Plutus.ChainIndex.Logging qualified as ChainIndex.Logging
-import Plutus.ChainIndex.Types (Point (..))
+import Plutus.ChainIndex.Config qualified as ChainIndex
+import Plutus.ChainIndex.Logging (defaultConfig)
 import Servant.Client (BaseUrl (BaseUrl), Scheme (Http))
-import System.Directory (createDirectory, doesFileExist, findExecutable)
+import System.Directory (copyFile, findExecutable)
 import System.Environment (setEnv)
 import System.Exit (die)
 import System.FilePath ((</>))
-import Test.Integration.Faucet (
-  genRewardAccounts,
-  maryIntegrationTestAssets,
-  mirMnemonics,
-  shelleyIntegrationTestFunds,
- )
 import Test.Plutip.Internal.BotPlutusInterface.Setup qualified as BotSetup
-import Test.Plutip.Internal.Types (ClusterEnv (..), RunningNode (RunningNode))
+import Test.Plutip.Internal.Types (
+  ClusterEnv (ClusterEnv, chainIndexUrl, networkId, runningNode, supportDir, tracer),
+  RunningNode (RunningNode),
+ )
+import Test.Plutip.Tools.CardanoApi qualified as Tools
 import UnliftIO.Concurrent (forkFinally)
 import UnliftIO.STM (TVar, atomically, newTVarIO, readTVar, retrySTM, writeTVar)
 
+import Data.Default (def)
+import Data.Foldable (for_)
+import GHC.Stack.Types (HasCallStack)
+import Paths_plutip (getDataFileName)
+import Test.Plutip.Config (PlutipConfig (chainIndexPort, clusterDataDir, relayNodeLogs))
+import Text.Printf (printf)
+import UnliftIO.Exception (catchIO)
+
 withPlutusInterface :: forall (a :: Type). ReaderT ClusterEnv IO a -> IO a
-withPlutusInterface action = withPlutusInterface' (runReaderT action)
+withPlutusInterface action = withPlutusInterface' def (runReaderT action)
```

Comment:
I think it would be cleaner, if we just required the `PlutipConfig` everywhere, maybe except for `withCluster`